tooling/gatsby-plugin/src/provider.js

```diff
@@ -0,0 +1,36 @@
+import * as React from "react"
+import {
+  ThemeProvider,
+  ColorModeProvider,
+  CSSReset,
+  GlobalStyle,
+  PortalManager,
+  extendTheme,
+} from "@chakra-ui/react"
+import userTheme from "./theme"
+
+const theme = extendTheme(userTheme)
+
+export const WrapRootElement = ({
+  element,
+  isResettingCSS,
+  isUsingColorMode,
+  portalZIndex,
+}) => {
+  const content = (
+    <>
+      {isResettingCSS && <CSSReset />}
+      <GlobalStyle />
+      <PortalManager zIndex={portalZIndex}>{element}</PortalManager>
+    </>
+  )
+  return (
+    <ThemeProvider theme={theme}>
+      {isUsingColorMode ? (
+        <ColorModeProvider options={theme.config}>{content}</ColorModeProvider>
+      ) : (
+        content
+      )}
+    </ThemeProvider>
+  )
```

Comment:
If `WrapRootElement` accepted the `ChakraProvider` props, this could become:

```suggestion
  return (
    <ChakraProvider theme={theme} {...props}>
      {element}
    </ChakraProvider>
  )
```

Reply:
Sounds good, I'll incorporate it!

Reply:
Mhh, but the `ChakraProvider` doesn't have a `isUsingColorMode` prop/option. This would be a breaking change then